File: backend/cografi-nesne-takip/src/main/java/com/kutalmis/cografi_nesne_takip/entity/BinType.java
package com.kutalmis.cografi_nesne_takip.entity;

public enum BinType {
    GENERAL("Genel"),
    RECYCLING("Geri Dönüşüm"),
    ORGANIC("Organik");

    private final String displayName;

    BinType(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}

File: frontend/lib/models/enum/bin_type.dart
enum BinType {
  general('Genel', 'GENERAL'),
  recycling('Geri Dönüşüm', 'RECYCLING'),
  organic('Organik', 'ORGANIC');

  final String displayName;
  final String apiValue;

  const BinType(this.displayName, this.apiValue);

  static BinType fromApiValue(String value) {
    return BinType.values.firstWhere(
      (e) => e.apiValue == value,
      orElse: () => throw ArgumentError('Bilinmeyen Cisim Türü: $value'),
    );
  }
}
